htdocs/product/stock/tpl/stockcorrection.tpl.php

```diff
@@ -128,26 +144,34 @@ function init_price()
 }
 
 // Serial / Eat-by date
-if (!empty($conf->productbatch->enabled) &&
+if (ismodEnabled('productbatch') &&
 (($object->element == 'product' && $object->hasbatch())
 || ($object->element == 'stock'))
 ) {
 	print '<tr>';
 	print '<td'.($object->element == 'stock' ? '' : ' class="fieldrequired"').'>'.$langs->trans("batch_number").'</td><td colspan="3">';
-	print '<input type="text" name="batch_number" class="minwidth300" value="'.GETPOST("batch_number").'">';
+	if ($pdluoid > 0) {
+		// If form was opened for a specific pdluoid, field is disabled
+		print '<input type="text" name="batch_number_bis" size="40" disabled="disabled" value="'.(GETPOST('batch_number') ?GETPOST('batch_number') : $pdluo->batch).'">';
+		print '<input type="hidden" name="batch_number" value="'.(GETPOST('batch_number') ?GETPOST('batch_number') : $pdluo->batch).'">';
+	} else {
+		print '<input type="text" name="batch_number" class="minwidth300" value="'.(GETPOST('batch_number') ? GETPOST('batch_number') : $pdluo->batch).'">';
```

Comment:
If we submit form with the field empty and we remain on page due to error, input fieds must remains as submitted, so should remain empty, shouldn't it ?

Reply:
![image](https://user-images.githubusercontent.com/3624836/184840041-8f2dec31-89e5-423d-aa1e-275905a3665f.png)